File: android/src/main/java/com/alphay/flutter/plugin/faceai/flutter_faceai/config/FaceImageConfig.java
package com.alphay.flutter.plugin.faceai.flutter_faceai.config;

public class FaceImageConfig {
    private String cacheSearchFaceDir;

    private static volatile FaceImageConfig instance;

    private FaceImageConfig() {}

    public static FaceImageConfig getInstance() {
        if (instance == null) {
            synchronized (FaceImageConfig.class) {
                if (instance == null) {
                    instance = new FaceImageConfig();
                }
            }
        }
        return instance;
    }

    public String getCacheSearchFaceDir() {
        return cacheSearchFaceDir;
    }

    public void setCacheSearchFaceDir(String cacheSearchFaceDir) {
        this.cacheSearchFaceDir = cacheSearchFaceDir;
    }
}

File: android/src/main/kotlin/com/alphay/flutter/plugin/faceai/flutter_faceai/config/FaceImageConfig.kt
package com.alphay.flutter.plugin.faceai.flutter_faceai.config

class FaceImageConfig private constructor() {

    var cacheSearchFaceDir: String? = null

    companion object {
        @Volatile
        private var instance: FaceImageConfig? = null

        fun getInstance(): FaceImageConfig {
            return instance ?: synchronized(this) {
                instance ?: FaceImageConfig().also { instance = it }
            }
        }
    }
}
